apps/api/v2/src/modules/slots/controllers/slots.controller.ts

```diff
@@ -129,6 +158,44 @@ export class SlotsController {
       },
     });
 
+    if (query.formatAsStartAndEndTime) {
+      let duration = 0;
+      if (query.duration) {
+        duration = query.duration;
+      } else if (query.eventTypeId) {
+        const eventType = await this.slotsService.getEventTypeWithDuration(query.eventTypeId);
+
+        if (!eventType) {
+          throw new Error("Event type not found");
+        }
+
+        duration = eventType.length;
+      } else {
+        throw new Error("duration or eventTypeId is required");
+      }
+
+      const transformedSlots = Object.entries(availableSlots.slots).reduce<
+        Record<string, { startTime: string; endTime: string }[]>
```

Comment:
we need to make sure availableSlots.slots is always defined and is an object. If not, the Object.entries(availableSlots.slots) call may throw an error.

```suggestion
     if (!availableSlots || !availableSlots.slots) {
         throw new Error("Available slots are missing"); 
      }
      const transformedSlots = Object.entries(availableSlots.slots).reduce<
        Record<string, { startTime: string; endTime: string }[]>
```

Reply:
It is already being handled by `getAvailableSlots`, it always returns slots, ref: https://github.com/calcom/cal.com/blob/4eac6c755dace7ae2c742c0d5e3a88283acd3942/packages/trpc/server/routers/viewer/slots/util.ts#L790